### DISCUSSION TITLE
'Constructors of embedded objects must not have required parameters'

### DISCUSSION THREAD

**Discussion by Sockitos** [isar/isar#581] [Help] [answered]
Hi,

I'm testing the latest dev version (isar 3.0.0-dev.10) and I encountered the following error while using the generator.
> Constructors of embedded objects must not have required parameters.

Is this a limitation that will still exist in the release version? 
Does this mean you can't have non-nullable properties in embedded classes?

Thanks!

**Comment by simc** [comment, marked as answer]
You can have non-nullable parameters but then you need to provide a default value in the constructor. This limitation will likely stay in stable v3.

**Reply by Mayb3Nots** [reply]
Hi @leisim may i know why this is a limitation? before 3.0 I could have nullable nested object in my collection

**Reply by simc** [reply]
Embedded objects were introduced in 3.0.

You can make them nullable but you need to provide a default constructor without required parameters so Isar can construct the object.

**Reply by Mayb3Nots** [reply]
> Embedded objects were introduced in 3.0.
> 
> You can make them nullable but you need to provide a default constructor without required parameters so Isar can construct the object.

My bad I was confused with Links vs embedded. Just to be clear we should use embedded in replacement for a Map?

**Reply by simc** [reply]
Not sure what you mean. Isar never supported Maps

**Reply by Mayb3Nots** [reply]
Yes so is embedded a solution for support maps? If not then whats the point of embedded?
